### PR TITLE
ISSUE 590 - IFC polylines

### LINKED ISSUE #590: Support for viewing 2D lines in 3D Repo when uploading .ifc models
### Description

Customer has noted that 2D lines are missing in 3D Repo when uploading the .ifc model (path to .ifc file below). This is expected as we currently do not export 2D lines from IFC.

File: [Track Design.ifc](https://asitesol.sharepoint.com/:u:/s/3DRepoTeamFolder/EbY-vccZj_VBuOajLCLMPLEBqFJ_-I4mWxxEv4xF_J4ITA?e=7cwd16)

We're using https://github.com/IfcOpenShell/IfcOpenShell for reading IFC files and their API allows provides us lines information - so this should be a relatively simple exercise of trying to get more data when we [extract geometry from the file](https://github.com/3drepo/3drepobouncer/blob/master/ifcUtils/repo_ifc_utils_geometry_handler.cpp#L122) 

### Goals
- [ ] As a user, I want to be able to view 2D lines from .ifc models in 3D Repo

### PR DESCRIPTION
**Pull request by sbanethia**
ISSUE #590 Changes for viewing 2D lines in 3Drepo when and IFC file is uploaded.

This fixes #590 

#### Description
By this changes user can view 2D lines in the 3drepo when he uploads IFC file.

### CODE REVIEW

**Review by carmenfan — CHANGES_REQUESTED** [review]
Trying it on other models - the lines don't seem to be at the right place:
image

image

image


You can find the files here: https://asitesol.sharepoint.com/:f:/s/3DRepoTeamFolder/EuVtlg-FYNJNkpMTZqV8lbcBfZCSotxkHklEJyqa4SO6cQ?e=WXZR9P

File: ifcUtils/repo_ifc_utils_geometry_handler.cpp [hunk 1]
```diff
@@ -101,6 +101,8 @@ bool repo::ifcUtility::SCHEMA_NS::GeometryHandler::retrieveGeometry(
 	std::vector<double>		&offset,
 	std::string              &errMsg)
 {
+	constexpr int data3D = 3;
```
[the remark this answers is not in the document]
Comment (sbanethia) [reply ifcUtils/repo_ifc_utils_geometry_handler.cpp:104]: Changes done.

File: ifcUtils/repo_ifc_utils_geometry_handler.cpp [hunk 2]
```diff
@@ -128,11 +130,12 @@ bool repo::ifcUtility::SCHEMA_NS::GeometryHandler::retrieveGeometry(
 		auto ob_geo = static_cast<const IfcGeom::TriangulationElement<double>*>(ob);
 		if (ob_geo)
 		{
-			auto primitive = 3;
+			auto primitive = data3D;
 			auto faces = ob_geo->geometry().faces();
+			auto edges = ob_geo->geometry().edges();
```
[the remark this answers is not in the document]
Comment (sbanethia) [reply ifcUtils/repo_ifc_utils_geometry_handler.cpp:135]: Changes done.

File: ifcUtils/repo_ifc_utils_geometry_handler.cpp [hunk 3]
```diff
@@ -227,8 +235,8 @@ bool repo::ifcUtility::SCHEMA_NS::GeometryHandler::retrieveGeometry(
 				}
 
 				post_faces[matInd].push_back(face);
-
-				if (primitive == 3) ++matIndIt;
+				
+				if ((primitive == data3D) || (primitive==data2D)) ++matIndIt;
```
[the remark this answers is not in the document]
Comment (sbanethia) [reply ifcUtils/repo_ifc_utils_geometry_handler.cpp:239]: Changes done.

File: ifcUtils/repo_ifc_utils_geometry_handler.cpp [hunk 4]
```diff
@@ -168,7 +171,7 @@ bool repo::ifcUtility::SCHEMA_NS::GeometryHandler::retrieveGeometry(
 
 			for (int iface = 0; iface < faces.size(); iface += primitive)
 			{
-				auto matInd = primitive == 3 ? *matIndIt : ob_geo->geometry().materials().size();
+				auto matInd = primitive == data3D ? *matIndIt : ob_geo->geometry().materials().size();
```
[the remark this answers is not in the document]
Comment (sbanethia) [reply ifcUtils/repo_ifc_utils_geometry_handler.cpp:174]: Changes done.

**Review by carmenfan — APPROVED** [review]

**Pull request merged**